src/main/scala/org/camunda/feel/impl/builtin/StringBuiltinFunctions.scala

```diff
@@ -95,7 +96,11 @@ object StringBuiltinFunctions {
     params = List("input", "pattern", "replacement"),
     invoke = {
       case List(ValString(input), ValString(pattern), ValString(replacement)) =>
-        ValString(input.replaceAll(pattern, replacement))
+        Try (Pattern.compile(pattern))
+          .map {pattern =>
+            val m = pattern.matcher(input)
+            ValString(m.replaceAll(replacement))
+          }.getOrElse(ValNull)
```

Comment:
What happens if you use a context as the pattern? Seems there is only a case for pattern as string. Is that guarded by the builtinFunction?

Reply:
Yes. The built-in functions have a guard for illegal arguments: https://github.com/camunda/feel-scala/blob/master/src/main/scala/org/camunda/feel/impl/builtin/BuiltinFunction.scala#L18-L25